### PR TITLE
[Dsahboard] Feature: Billing V2 UI changes

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on refactoring billing-related components and types in the application, enhancing the onboarding process, and improving the pricing structure for different user plans. It also includes the removal of deprecated files and introduces new functionalities for managing billing and account settings.

### Detailed summary
- Deleted several unused files related to onboarding and billing.
- Refactored `TrackedLinkButtonProps` to simplify its interface.
- Updated `billingPlan` to include a new "starter" option.
- Modified `teamStub` function to use the updated billing plan type.
- Renamed `LazyOnboardingBilling` to `LazyAddPaymentMethod`.
- Changed `OnboardingModal` to `TWModal` with updated props.
- Improved account fetching logic in `getAccount`.
- Refactored `TeamPlanBadge` to use the new billing plan type.
- Simplified the logic for determining valid team plans in `getValidTeamPlan`.
- Enhanced `CreditsItem` and `BillingPricing` components for better display and management of billing information.
- Updated the `PricingSection` to reflect changes in billing plans and CTA links.
- Introduced `PlanInfoCard` to display account and usage billing details effectively.
- Removed unnecessary state management related to payment methods in several components.

> The following files were skipped due to too many changes: `apps/dashboard/src/components/homepage/sections/PricingCard.tsx`, `apps/dashboard/src/components/settings/Account/Billing/index.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5436?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5436** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5436?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5436?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#5432** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5432?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5430** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5430?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: apps/dashboard/src/components/onboarding/ChoosePlan.tsx [hunk 1]
```diff
@@ -1,118 +1,61 @@
...
-            variant: "solid",
-            colorScheme: "blue",
+            // directly redirect to stripe on onboarding
+            href: "/team/~/subscribe/plan:growth",
+            variant: "default",
           }}
-          onDashboard
+          // Question: @Jonas
+          // Is this true? - all new users can try grow for 1 month?
+          canTrialGrowth={true}
```
Comment (MananTank) [inline apps/dashboard/src/components/onboarding/ChoosePlan.tsx:54]: @jnsdls question for you here

File: apps/dashboard/src/components/settings/Account/Billing/Pricing.tsx [hunk 2]
```diff
@@ -1,125 +1,173 @@
...
-  validPayment: boolean;
-  paymentVerification: boolean;
-  invalidPayment: boolean;
-  loading: boolean;
-  onSelect: (plan: AccountPlan) => void;
 }
 
+// Question: @Jonas
+//  I'm disregarding the plan info on the Account, and using the Team as source of truth - is that ok?
+
```
Comment (MananTank) [inline apps/dashboard/src/components/settings/Account/Billing/Pricing.tsx:17]: @jnsdls question for you

File: apps/dashboard/src/utils/pricing.tsx [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
 
 export const CONTACT_US_URL =
   "https://meetings.hubspot.com/sales-thirdweb/thirdweb-pro";
 
-export const PLANS: {
-  [T in AccountPlan]: {
+type NonFreeTeamPlan = Exclude<Team["billingPlan"], "free">;
+
+// Question: @Jonas is this correct?
+export const TEAM_PLANS: Record<
```
Comment (MananTank) [inline apps/dashboard/src/utils/pricing.tsx:10]: @jnsdls question for you. Is the features listed on each plan still correct?

File: apps/dashboard/src/components/onboarding/ChoosePlan.tsx (hunk 1 above)
Comment (jnsdls) [reply apps/dashboard/src/components/onboarding/ChoosePlan.tsx:54]: if they have never had growth plan before then yes - I'll update this to be included in the api response so we can drive it from that

File: apps/dashboard/src/components/settings/Account/Billing/Pricing.tsx (hunk 2 above)
Comment (jnsdls) [reply apps/dashboard/src/components/settings/Account/Billing/Pricing.tsx:17]: yes this is correct

File: apps/dashboard/src/utils/pricing.tsx (hunk 3 above)
Comment (jnsdls) [reply apps/dashboard/src/utils/pricing.tsx:10]: for now lets say yes

File: apps/dashboard/src/components/onboarding/ChoosePlan.tsx (hunk 1 above)
Comment (MananTank) [reply apps/dashboard/src/components/onboarding/ChoosePlan.tsx:54]: I see the usage of `!account.trialPeriodEndedAt` for determining  this - does that work?

**Pull request merged**